5_evaluate/src/eval_utility_fxns.R

```diff
@@ -53,9 +56,27 @@ match_pred_obs <- function(preds_file, eval_obs, driver) {
     select(-ice) %>%
     munge_long() %>% # Fxn in 4_visualize/src/plot_data_utility_fxns.R
     mutate(depth = as.numeric(depth)) %>% 
-    rename(pred = temperature) %>%
-    arrange(time, depth)
+    rename(pred = temperature)
   
+  # If lake_depth exceeds the maximum GLM prediction depth
+  if (lake_depth$lake_depth > max(eval_preds$depth)) {
+    # Generate a new dataframe with a row for maximum lake depth for each date
+    new_depths <- tibble(
+      time = unique(eval_preds$time),
+      site_id = lake_depth$site_id,
+      depth = lake_depth$lake_depth
+    )
+    eval_preds <- eval_preds %>%
+      bind_rows(new_depths)
+  }
+  
+  # Fill predictions to the maximum GLM prediction depth or to the maximum lake depth,
+  # if it exceeds the maximum GLM prediction depth
+  eval_preds <- eval_preds %>%
+    arrange(time, depth) %>%
+    fill(pred, .direction=c('down'))
```

Comment:
Wow, I have never seen `fill()` before. You always find these handy little functions!! 

Putting this example here for future us in case we are wondering what this function does. This is the little example I used to understand what is going on here. 

```r
eval_preds <- tibble(
  time = c(rep(1,3), rep(2,2)), 
  site_id = rep("site1", 5),
  depth = c(0:2, 0:1), 
  pred = c(7.6, 5.4, 3.3, 7.8, 6.2))

new_depths <- tibble(
  time = c(1,2),
  site_id = "site1",
  depth = 3
)

# NAs appear for `preds` for the new depths once you bind them
eval_preds <- eval_preds %>% 
  bind_rows(new_depths)

# See just this first step when you arrange by time & depth
# Both new depths still have NAs
eval_preds %>% 
  arrange(time, depth)

# A tibble: 7 x 4
   time site_id depth  pred
  <dbl> <chr>   <dbl> <dbl>
1     1 site1       0   7.6
2     1 site1       1   5.4
3     1 site1       2   3.3
4     1 site1       3  NA  
5     2 site1       0   7.8
6     2 site1       1   6.2
7     2 site1       3  NA  

# `fill(..., .direction='down')` inserts the value from above each NA 
eval_preds %>% 
  arrange(time, depth) %>%
  fill(pred, .direction=c('down'))

# A tibble: 7 x 4
   time site_id depth  pred
  <dbl> <chr>   <dbl> <dbl>
1     1 site1       0   7.6
2     1 site1       1   5.4
3     1 site1       2   3.3
4     1 site1       3   3.3
5     2 site1       0   7.8
6     2 site1       1   6.2
7     2 site1       3   6.2
```

Reply:
Thanks, Lindsay! Definitely helpful to have this example documented.